pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -523,7 +519,10 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 		}
 		selectedNode := pvc.Annotations[pvutil.AnnSelectedNode]
 		if selectedNode != pod.Spec.NodeName {
-			return false, fmt.Errorf("selectedNode annotation value %q not set to scheduled node %q", selectedNode, pod.Spec.NodeName)
```

Comment:
IMO, the original error msg here may be better

Reply:
See https://github.com/kubernetes/kubernetes/issues/83181 and discussion in pr #83343

Reply:
Sorry, I see the issue but missed the pr.

Looks good. Users should not care the annotation. :)